app/src/main/java/org/mozilla/focus/IntentValidator.kt

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.TextUtils
+import org.mozilla.focus.session.Source
+import org.mozilla.focus.utils.SafeIntent
+import org.mozilla.focus.utils.UrlUtils
+
+typealias OnValidBrowserIntent = (url: String, source: Source) -> Unit
+
+/**
+ * A container for functions that parse Intents and notify the application of their validity.
+ *
+ * This class uses callbacks to notify the application in order to decouple the Intent parsing
+ * from the application logic, which also allows us to test the functionality.
+ *
+ * The functions in this class take [SafeIntent] in order to encourage the caller to use SafeIntents
+ * in their code.
+ */
+object IntentValidator {
+
+    fun validateOnCreate(context: Context, intent: SafeIntent, savedInstanceState: Bundle?, onValidBrowserIntent: OnValidBrowserIntent) {
```

Comment:
\[detekt\] INFO: \[style/MaxLineLength\] Line detected that is longer than the defined maximum line length in the code style\.

Reply:
Sputnik isn't reading our detektConfig, unfortunately, but this doesn't fail locally.